Corion/upstream cygwin hints

Corion opened the pull request:
I've had this PR laying around on my local Git repo for a long time. This simply upstreams some patches that Cygwin applies to Perl.

jkeenan commented:
> I've had this PR laying around on my local Git repo for a long time. This simply upstreams some patches that Cygwin applies to Perl.

Currently no one is submitting smoke-test reports from Cygwin.  We have one GitHub Action set up for Cygwin, but I don't know how reliable or relevant it is.  How can we go about QA-ing this pull request?

Corion commented:
This is a good question. I'm mostly the messenger of the downstream patches, but I don't know how to test them. I'll investigate whether there is a Cygwin maintainer that can tell us.

demerphq commented:
@Corion this is out of data and conflicts now, is it out of date or should we get it rebased and apply it?

Commit:
Upstream Cygwin hints patches

From

https://cygwin.com/git-cygwin-packages/?p=git/cygwin-packages/perl.git;a=blob_plain;f=perl.cygwin-hints.patch;hb=de363db36fa754fcee9e07d13aec1eed571cf0dc

# Conflicts:
#	hints/cygwin.sh

Corion commented:
Rebased - I haven't had contact with any Cygwin maintainer, so I still don't know how to properly test these changes ...

I guess I'll try to download mingw64 and do a Perl build on Windows

demerphq commented on hints/cygwin.sh:
```diff
@@ -71,13 +71,23 @@ case "$osvers" in
   static_ext="$static_ext Win32CORE"
 fi
 
+# configury should not check for xlocale.h if it gets the API from locale.h
```
typo on "configury"? should that be "configure"?

Corion replied:
... it's now fixed

demerphq replied:
cool. thanks.

Corion commented:
I just built a Perl on Cygwin with the changes from this branch, and the tests passed (except Net::Ping, which I blame on things unrelated to this change, and two porting/ tests regarding file attributes.

So IMO this can be merged.

sisyphus commented:
According to google, cygwin-2.5.0 was released on June 23, 2016.
IIUC, this patch affects only those Cygwin versions that are older than 2.5.0.

On my current cygwin installation (version 3.3.6) I applied, by hand, this hints/cygwin/cygwin.sh patch to current bleadperl (commit fdce04e) and ran:
```
$ sh Configure -des -Dusedevel -Uversiononly -Dusethreads -Dusemultiplicity -Dprefix=/cygdrive/d/cygperl-5.37.891-ld -Duselongdouble -Dlibpth=/lib/gcc/x86_64-pc-cygwin/11 && make && make test
```
The test report contained:
```
Failed 3 tests out of 2594, 99.88% okay.
        ../dist/Net-Ping/t/450_service.t
        ../lib/File/Copy.t
        porting/authors.t
```
Making amendments to a file by hand often buggers up the file permissions.
Not sure about the other failing tests (Net-Ping and File-Copy) but I think they're a regular feature of Cygwin builds.
It's pretty late over here, right now - I can check more thoroughly in the morning.

Cheers,
Rob

Commit:
Fix typo

sisyphus commented:
> I can check more thoroughly in the morning.

I agree - this alteration to cygwin.sh has nothing to do with the failing tests.

Cheers,
Rob
